src/software/new_geom/polynomial.cpp

```diff
@@ -33,7 +33,7 @@ unsigned int Polynomial::getOrder() const
 {
     if (coeffs.size() != 0)
     {
-        for (size_t i = coeffs.size() - 1; i >= 0; i++)
+        for (size_t i = coeffs.size() - 1; i >= 0; i--)
```

Comment:
Could we add a unit test that fails without this change to make sure we don't regress in the future?

Reply:
Added `get_order` test